sequencer/src/persistence/fs.rs

```diff
@@ -147,6 +177,169 @@ impl SequencerPersistence for Persistence {
         Ok(Some(bincode::deserialize(&bytes).context("deserialize")?))
     }
 
+    async fn load_high_qc(&self) -> anyhow::Result<Option<QuorumCertificate<SeqTypes>>> {
+        let path = self.high_qc_path();
+        if !path.is_file() {
+            return Ok(None);
+        }
+        let mut file = File::open(path)?;
+
+        file.seek(SeekFrom::Start(8)).context("seek")?;
+        let bytes = file
+            .bytes()
+            .collect::<Result<Vec<_>, _>>()
+            .context("read")?;
+        Ok(Some(bincode::deserialize(&bytes).context("deserialize")?))
+    }
+
+    async fn load_da_proposal(
+        &self,
+        view: ViewNumber,
+    ) -> anyhow::Result<Option<Proposal<SeqTypes, DAProposal<SeqTypes>>>> {
+        let dir_path = self.da_dir_path();
+
+        if !dir_path.exists() {
+            return Ok(None);
+        }
+
+        let entries = fs::read_dir(dir_path)?;
+        let view = view.get_u64();
+
+        for entry in entries {
```

Comment:
Since we know the view number, just figure out the file name and open it directly, instead of searching